src-tauri/src/systemtray_manager.rs

```diff
@@ -0,0 +1,239 @@
+            SystrayItemId::CpuHashrate.to_str(),
+            SystrayItemId::CpuHashrate.get_title(0.0),
+            false,
+            None::<&str>,
+        )?;
+        let gpu_hashrate = MenuItem::with_id(
+            &app,
+            SystrayItemId::GpuHashrate.to_str(),
+            SystrayItemId::GpuHashrate.get_title(0.0),
+            false,
+            None::<&str>,
+        )?;
+        let estimated_earning = MenuItem::with_id(
+            &app,
+            SystrayItemId::EstimatedEarning.to_str(),
+            SystrayItemId::EstimatedEarning.get_title(0.0),
+            false,
+            None::<&str>,
+        )?;
+        let minimize_toggle = MenuItem::with_id(
+            &app,
+            SystrayItemId::MinimizeToggle.to_str(),
+            SystrayItemId::MinimizeToggle.get_title(0.0),
+            true,
+            None::<&str>,
+        )?;
+
+        let menu = Menu::with_items(
+            &app,
+            &[
+                #[cfg(target_os = "macos")]
+                &about,
+                #[cfg(target_os = "macos")]
+                &separator,
+                &cpu_hashrate,
+                &gpu_hashrate,
+                &separator,
+                &estimated_earning,
+                &separator,
+                &minimize_toggle,
+            ],
+        )?;
+        Ok(menu)
+    }
+
+    fn get_tooltip_text(&self, data: SystemTrayData) -> String {
+        match PlatformUtils::detect_current_os() {
+            CurrentOperatingSystem::Linux => "Not supported".to_string(),
+            _ => {
+                format!(
+                    "CPU Power: {}\nGPU Power: {}\nEst. earning: {}",
+                    format_hashrate(data.cpu_hashrate),
+                    format_hashrate(data.gpu_hashrate),
+                    format_currency(data.estimated_earning / 1_000_000.0, "tXTM/day")
+                )
+            }
+        }
+    }
+
+    pub fn initialize_tray(&mut self, app: AppHandle) -> Result<(), anyhow::Error> {
+        let tray = app.tray_by_id("universe-tray-id").expect("tray not found");
+        let menu = self.initialize_menu(app.clone())?;
+        tray.set_menu(Some(menu.clone()))?;
+
+        tray.on_menu_event(move |app, event| match event.id.as_ref() {
+            "minimize_toggle" => {
+                let window = match app.get_webview_window("main") {
+                    Some(window) => window,
+                    None => {
+                        error!(target: LOG_TARGET, "Failed to get main window");
+                        return;
+                    }
+                };
+
+                if window.is_minimized().unwrap_or(false) {
+                    info!(target: LOG_TARGET, "Unminimizing window");
+                    match PlatformUtils::detect_current_os() {
+                        CurrentOperatingSystem::Linux => {
+                            window.hide().unwrap_or_else(|error| error!(target: LOG_TARGET, "Failed hide window: {}", error));
```

Comment:
We have to hide it before we unminimize it?

Reply:
`window` works really bad on linux. I played with this a lot, it won't never unminimize using the mac/windows scenario